### DISCUSSION TITLE
Signing via Jsign seemingly successful, SignTool verify fails, on ClearCase MVFS

### DISCUSSION THREAD

**Discussion by LanceHall01** [ebourg/jsign#207] [Q&A] [answered]
I have a set of 10 EXEs that are created during our product build via the WIX installer utility. All of these EXEs are then signed/timestamped with JSign 6.0 which gives no indication of failure. I then run these signed EXEs through SignTool verify which fails some of the EXEs randomly - anywhere from 3 to 9 of the 10 fail. Each time when our product build creates the 10 EXEs anew, the subset that fails Signtool verify changes. The Windows properties of the failed file shows "This digital signature is not valid" for the Digital Signature Details. Overall, there are a couple hundred files that are signed/timestamped via JSign during our product build but only these 10 EXEs are randomly problematic.

Questions:

1. Is there some kind of JSign option to increase its logging verbosity that I cannot seemingly find?
2. What else can I submit here so that the community can help?

Thanks!!!

**Comment by ebourg** [comment]
Could you send a file with an invalid signature to ebourg@apache.org please? I'll get a look.

**Comment by ebourg** [comment, marked as answer]
Long story short, after analyzing the file sent by Lance, it appears that padding the PE file before writing the certificate table has no effect on a ClearCase MVFS filesystem. The workaround found is to write the certificate table directly at the right location without padding first (1942f9f76886c1bdc94a42f1d71472b00e1cd7f3).
